libs/ngx-charts-on-fhir/src/lib/fhir-chart-summary/medication-summary.service.ts

```diff
@@ -12,6 +13,9 @@ export class MedicationSummaryService implements SummaryService {
     return layer.scale.type === 'category';
   }
   summarize(layer: DataLayer<TimelineChartType, MedicationDataPoint[]>): Record<string, string>[] {
+    const groups = groupBy(layer.datasets, (dataset) => dataset.chartsOnFhir?.group ?? dataset.label);
+    const datasets = Object.entries(groups).map(([label, ds]) => ({ ...ds[0], label, data: ds.flatMap((d) => d.data) }));
+    layer = { ...layer, datasets };
     return layer.datasets.map((dataset) => ({
```

Comment:
reassigning layer is unnecessary. You can just use the datasets var:
```suggestion
    return datasets.map((dataset) => ({
```